### ISSUE TITLE
Error 403 with zrok and other reverse proxies

### ISSUE THREAD

**Issue by freQuensy23-coder** [ollama/ollama#3269]
### What is the issue?

After updating to 1.29, I lost the ability to make public access to the ollama instance I have running through ngrok (or analogues (like zrok)). Ollama returns a 403 response to requests received through a proxy (ngrok) while correctly responding to the request through the localhost

### What did you expect to see?

Ngrok should work

### Steps to reproduce

Install newest version of ollama
Install ngrok/zrok
zrok share public localhost:11434

### OS

Linux

### Architecture

x86

### Ollama version

1.29

**Comment by ftoppi** [comment]
Did you try command in the [faq](https://github.com/ollama/ollama/blob/main/docs/faq.md)?
`ngrok http 11434 --host-header="localhost:11434"`

**Comment by projetosfsi** [comment]
Same Problem with CloudFlare Tunnel

I downgraded Ollama to version 1.2.8 and it worked again

**Comment by jmorganca** [comment]
Hi @freQuensy23-coder and @projetosfsi I've added sections in the FAQ on using Ollama with tunneling tools:
* [Ngrok](https://github.com/ollama/ollama/blob/main/docs/faq.md#how-can-i-use-ollama-with-ngrok)
* [Cloudflare Tunnel](https://github.com/ollama/ollama/blob/main/docs/faq.md#how-can-i-use-ollama-with-cloudflare-tunnel)

Hope this helps!

**Comment by josh** [comment]
Just hit the same issue using [Tailscale's serve command](https://tailscale.com/kb/1242/tailscale-serve). Unfortunately it doesn't have an easy flag to rewrite the Host header.

**Comment by jmorganca** [comment]
Hi @quantumalchemy @josh  sorry you hit this. I'll work on fixing the host verification so this doesn't get in the way.

In the meantime, for Zrok, you can use their new Caddy implementation with a Caddyfile like this:

```
http:// {
    bind {{ .ZrokBindAddress }}
    reverse_proxy 127.0.0.1:113434 {
        header_up Host localhost:11434
    }
}
```

See https://blog.openziti.io/zrok-with-the-power-of-caddy

**Comment by alexander-potemkin** [comment]
Faced same issue with pure Caddy - thanks for the `header_up` thing!

**Comment by melMass** [comment]
> Just hit the same issue using [Tailscale's serve command](https://tailscale.com/kb/1242/tailscale-serve). Unfortunately it doesn't have an easy flag to rewrite the Host header.

Did you ever found a solution using tailscale, I just get 403. I use serve with all kinds of local servers but never face this:

```sh
[GIN] 2024/08/18 - 22:17:04 | 403 |            0s |   <IP-REDACTED> | HEAD     "/"
[GIN] 2024/08/18 - 22:17:12 | 403 |            0s |   <IP-REDACTED> | HEAD     "/"
[GIN] 2024/08/18 - 22:17:16 | 403 |            0s |   <IP-REDACTED> | GET      "/"
[GIN] 2024/08/18 - 22:17:21 | 403 |            0s |   <IP-REDACTED> | GET      "/"
[GIN] 2024/08/18 - 22:19:22 | 403 |            0s |   <IP-REDACTED> | GET      "/"
[GIN] 2024/08/18 - 22:19:29 | 403 |            0s |   <IP-REDACTED> | GET      "/"
[GIN] 2024/08/18 - 22:19:50 | 403 |            0s |   <IP-REDACTED> | GET      "/"
```

**Comment by graciasc** [comment]
> Hi @quantumalchemy @josh sorry you hit this. I'll work on fixing the host verification so this doesn't get in the way.
> 
> In the meantime, for Zrok, you can use their new Caddy implementation with a Caddyfile like this:
> 
> ```
> http:// {
>     bind {{ .ZrokBindAddress }}
>     reverse_proxy 127.0.0.1:11434 {
>         header_up Host localhost:11434
>     }
> }
> ```
> 
> See https://blog.openziti.io/zrok-with-the-power-of-caddy

tldr:  For anyone who wants a quick answer

This is the command you would need to run to get it working for `zrok` 

`zrok share public --backend-mode caddy <file path to caddy file with the above>`

**Comment by jmorganca** [comment]
Thanks for the issue! For security setting the host header is required
